qemu/tests/qemu_guest_agent.py

```diff
@@ -1512,6 +1525,21 @@ def gagent_check_file_read(self, test, params, env):
         :param params: Dictionary with the test parameters
         :param env: Dictionary with test environment.
         """
+        def _read_guest_file_with_count(count_num):
+            """
+            Read a guest file with count number.
+
+            :params: count_num: read file with count number on demand.
+            """
+            try:
+                self.gagent.guest_file_read(ret_handle, count=int(count_num))
+            except guest_agent.VAgentCmdError as detail:
+                info_insuffi = "Insufficient system resources exist to"
+                info_insuffi += " complete the requested service"
+                if not re.search(info_insuffi, str(detail)):
```

Comment:
```suggestion
                if not info_insuffi in str(detail.edata):
```
or better specify the exact key of `detail.edata`, I guess it might be 'desc'(then it would be `if not info_insuffi in detail.edata['desc']`) but I am not sure, you might check it within the error detail.

Reply:
1. How to understand detail.edata, edata is a build-in function for string?
2. count for windows now should be reduce to 10000000,otherwise case will be fail. so what do you think?

Reply:
> 1. How to understand detail.edata, edata is a build-in function for string?

`except guest_agent.VAgentCmdError as detail` so `detail` is `guest_agent.VAgentCmdError` type, you might check the definition of it: https://github.com/avocado-framework/avocado-vt/blob/master/virttest/guest_agent.py#L64 Here the object has the property `self.edata`, and where raise the error : https://github.com/avocado-framework/avocado-vt/blob/master/virttest/guest_agent.py#L441 And, you could debug your log to check out the detail of the `detail`, list it's content, from the code that raise it I guess it should be a dictionary.

> 2. count for windows now should be reduce to 10000000,otherwise case will be fail. so what do you think?

`count = 1000000000 if params["os_type"] == 'linux' and main_qga_ver<5  else 10000000` So for windows it will go to the `else` so it is `10000000` as you expected, right?

Reply:
1. the error info from debug.log is : {'class': 'GenericError', 'desc': "value '1000000000' is invalid for argument count"})') ,so it should be  if not info_insuffi in detail.edata['desc'], you're right. that's more targeted, thanks qianqian, learn more.
2. yes, 10000000 is my expected value for windows as well ,once the value is bigger than 10000000, windows cases will be failed.